.idea/Working version/algorithms.js

```diff
@@ -0,0 +1,472 @@
+/**
+ * Created by tsodergren on 6/5/17.
+ */
+
+
+function constructEdges() {
+
+
+    var radiusChange = arguments.length ? arguments[0] : 0;
+
+    //squared coverage diameter and annulus of uncertainty
+    var sqDiameter = 4 * Math.pow(complexRadius, 2);
+    var sqDiameterMin = 4 * Math.pow(complexRadius-dataRadius, 2);
+    var sqDiameterMax = 4 * Math.pow(complexRadius+dataRadius, 2);
+
+
+    if (!a_edges) {
+        var x=math.zeros(numSamples, numSamples,'sparse');
+        a_edges = {edgeProb: x.clone(), edgeAnchorDist: x.clone(), edgeDist: x.clone(), edgeFlag: x.clone(),
+            edgeInd: x.clone() };
+    }
+
+    var iEdgeDist, iEdgeFlag, tempEdges;
+
+    if ( radiusChange==1 ) {
+        tempEdges = JSON.parse(JSON.stringify(ripsEdges));
```

Comment:
We could possibly use Object.assign (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/assign). The only downside with this is that it doesn't support Internet Explorer. I'm not sure if we wanted to make sure it worked on everything or add that stipulation.